opflexagent/gbp_ovs_agent.py

```diff
@@ -277,15 +277,68 @@ def port_bound(self, port, net_uuid,
                     port, net_uuid, network_type, physical_network,
                     segmentation_id, fixed_ips, device_owner, ovs_restarted)
         elif network_type in self.supported_pt_network_types:
+            # PT cleanup is needed before writing the new endpoint files
+            self.mapping_cleanup(port.vif_id, cleanup_vrf=False)
             if ((self.opflex_networks is None) or
                     (physical_network in self.opflex_networks)):
                 # Port has to be untagged due to a opflex agent requirement
                 self.int_br.clear_db_attribute("Port", port.port_name, "tag")
-                self.mapping_to_file(port, net_uuid, mapping, fixed_ips,
+                # Multiple files will be created based on how many MAC
+                # addresses are owned by the specific port.
+                mapping_copy = copy.deepcopy(mapping)
+                mac_aap_map = {}
+                mapping_copy['allowed_address_pairs'] = []
+                mapping_copy['floating_ip'] = []
+                fip_by_fixed = {}
+                original_mac = mapping.get('mac_address') or port.vif_mac
+                # Prepare FIPs by fixed_ip
+                for fip in mapping.get('floating_ip', []):
+                    fip_by_fixed.setdefault(
+                        fip['fixed_ip_address'], []).append(fip)
+                # For the main MAC, set floating IP collection to all those
+                # FIPs pointing to the Port fixed ips.
+                for fixed in fixed_ips:
+                    mapping_copy['floating_ip'].extend(
+                        fip_by_fixed.get(fixed['ip_address'], []))
+
+                # For the main MAC EP, set al the AAP with no mac address or
+                # MAC address equal to the original MAC.
+                for aap in mapping.get('allowed_address_pairs', []):
+                    if not aap.get('mac_address') or aap.get(
+                            'mac_address') == original_mac:
+                        # Should go with the MAIN mac address EP file
+                        mapping_copy['allowed_address_pairs'].append(aap)
```

Comment:
Yes I am. Please refer to the UTs added below (especially the multi file one, in which I pretty much cover all this).
Consider that multiple AAP may and up in the same file (AAP with same mac address) so you basically store them by MAC (one or multiple).